src/components/NavMenu.tsx

```diff
@@ -0,0 +1,92 @@
+import React, { useState, Fragment } from "react";
+import { NavLink } from "react-router-dom";
+import styled, { css } from "styled-components";
+import { useTranslation } from "react-i18next";
+import navButton from "./assets/placeholder-nav.jpeg";
+import { space, SpaceProps, TypographyProps, typography } from "styled-system";
+
+const visibleStyles = css`
+  display: flex;
+  flex-direction: column;
+  height: 100%;
+  background-color: rgba(0, 0, 0, 0.1);
+  font-size: 36px;
+  justify-content: center;
+  text-align: center;
+  width: 100%;
+  font-family: SangBleu OG Serif Light Regular;
+`;
+const NavMenuOverlay = styled.dialog<{ isOpen: boolean }>`
+  display: none;
+  ${(props) => props.isOpen && visibleStyles}
+`;
+
+const MenuButton = styled.button`
+  position: absolute;
+  right: 1em;
+  top: 1em;
+  height: 50px;
+  width: 50px;
+`;
+
+type MenuItemProps = SpaceProps & TypographyProps;
+
+const MenuItem = styled.li<MenuItemProps>`
+  ${space};
+  ${typography};
+
+  color: #000000;
```

Comment:
If we stick `height: 66px` here, then when the mouse hovers over a menu item, the rest of the items don't jump down (if that makes sense?)

Reply:
Orrr actually... (i've just seen that padding is controlled further down)... if we used styled-system `layout` function rather than `space`, and give heights for the menu items rather than padding, it would give the same effect.
However, for the text to still be centered we'd need the menu button `display:flex` `align-items:center` `justify-content:center`.

Reply:
Thanks Matt! I've added a `height: 66px` since the I couldn't find `layout` in the styled-system theme file. This height fix did the trick though.